src/Engine/UnitActionLifecycles/BattleLifecycle.java

```diff
@@ -169,6 +175,11 @@ public GameEventQueue getEvents(MapMaster gameMap)
           if( event.defenderDies() )
           {
             attackEvents.add(new UnitDieEvent(event.getDefender()));
+            if( event.getDefender().model.holdingCapacity > 0 )
+              for( Unit u : event.getDefender().heldUnits )
+              {
+                attackEvents.add(new UnitDieEvent(u));
+              }
```

Comment:
Would it be feasible to put this check into the `UnitDieEvent`, or maybe make a helper function to create the set of them?
I don't like the idea of having to track down units floating in the void after their transports get wrecked.

Reply:
We could certainly remove any transported units in the `UnitDieEvent`, but then they couldn't have their own death event, which I'd rather have.
A utility function would have to be somewhere accessible to both `BattleLifecycle` and `DeleteLifecycle`, and wouldn't save much syntax here anyway. This is a simple enough operation that I'm not sure I care to break it out.

Reply:
It's not about saving syntax; it's about ensuring program correctness through a consistent approach.
If I make a helper function that queues a death event for a unit and all the unit's cargo and all the *cargo's* cargo (which, I suppose I'll note, you haven't accounted for here), then I shouldn't ever have to call the `UnitDieEvent` constructor explicitly anywhere else, and I don't have to remember to check for ~~traps~~ edge cases.

Reply:
Hmm, yes. Okay. Done.